src/Avalonia.Controls/AppBuilderBase.cs

```diff
@@ -145,6 +145,15 @@ public void Start<TMainWindow>(TMainWindow mainWindow, Func<object> dataContextP
             Instance.Run(mainWindow);
         }
 
+        public delegate void AppMainDelegate(Application app, string[] args);
```

Comment:
Why define a delegate here rather than use `Action<Application, string[]>`? Personally I prefer to use `Action`, `Func` as intellisense tells you what the parameters are...

Reply:
Wanted the intellisense engine to insert actual parameter names instead of `arg1` and `arg2`

Reply:
Fair enough, i guess each method has its own pros/cons 👍